mlir/lib/Dialect/LLVMIR/IR/GENXOps.cpp

```diff
@@ -30,3 +30,32 @@ LogicalResult GENX::MatrixLoadOp::verify() {
 
   return success();
 }
+
+//===----------------------------------------------------------------------===//
+// genx.matrix.mad
+//===----------------------------------------------------------------------===//
+
+LogicalResult GENX::MatrixMadOp::verify() {
+  // The scope attribute must be 'Subgroup' currently.
+  if (getScope() != GENX::Scope::Subgroup)
+    return this->emitOpError("scope attribute must have value 'Subgroup'");
+
+  auto AType = getA().getType().cast<GENX::JointMatrixType>();
+  auto BType = getB().getType().cast<GENX::JointMatrixType>();
+  auto CType = getC().getType().cast<GENX::JointMatrixType>();
+  auto resType = getResult().getType().cast<GENX::JointMatrixType>();
+
+  if (CType != resType)
+    return this->emitOpError("result and 3rd operand must have the same type");
+
+  if (AType.getNumRows() != resType.getNumRows() ||
+      AType.getNumColumns() != BType.getNumRows() ||
+      BType.getNumColumns() != resType.getNumColumns())
+    return this->emitOpError("matrix sizes must match");
+
+  if (AType.getElementType() != BType.getElementType() ||
+      resType.getElementType() != CType.getElementType())
```

Comment:
we already checked `CType == resType`, let's check
```suggestion
      AType.getElementType() != CType.getElementType())
```

Reply:
Here I think we need to check that the type combination is valid according to allowed combinations in https://github.com/intel/llvm/blob/54cd00d559a89931e4bcc5cf0dbf5938e2767c52/sycl/doc/extensions/experimental/sycl_ext_matrix/sycl_ext_oneapi_matrix.asciidoc#intel-xmx-supported-combinations

Reply:
done and more verification test have been added